papers/source/d1664.bs

```diff
@@ -130,22 +130,28 @@ In short, this paper formalizes the work done in P1739 by giving it an expositio
 
 # Design # {#design}
 
-The concept is simple and is given in 2 exposition-only types added to the standard:
+The design is simple and is given in 2 exposition-only concepts added to the standard:
 
 ```
 template <typename R>
-concept pair-reconstructible-range = 
-    forwarding-range<std::remove_reference_t<R>> && 
-    std::Constructible<std::remove_reference_t<R>, iterator_t<R>, sentinel_t<R>>;
+concept pair-reconstructible-range =
+    Range<R> &&
+    forwarding-range<std::remove_reference_t<R>> &&
+    std::Constructible<R, iterator_t<R>, sentinel_t<R>>;
 
 template <typename R>
-concept range-reconstructible-range = 
-    forwarding-range<std::remove_reference_t<R>> && 
-    std::Constructible<std::remove_reference_t<R>, std::ranges::subrange<iterator_t<R>, sentinel_t<R>>>;
+concept range-reconstructible-range =
+    Range<R> &&
+    forwarding-range<std::remove_reference_t<R>> &&
+    std::Constructible<R, std::ranges::subrange<iterator_t<R>, sentinel_t<R>>>;

```

Comment:
We forgot to check for `Range<R>` and we don't want to remove_reference for the constructible check.

Reply:
We could also do `View<R>` instead of `Range<R>` which might represent the intent better, but technically it's not necessary, I think.

Reply:
Let's keep it at `Range<R>`. I don't want to restrict this too tightly.

Reply:
I am still undecided about this one. I don't think there are any ranges in the standard or in range-v3 or proposed that model `forwarding-range<std::remove_reference_t<R>>` that do not also model `View<R>`. And I also cannot think of why you would make a range whose entire state is part of the iterators *not* also model `View`. Also `View` expresses that we want (re)construction to be in O(1) which is not stated anywhere in our standard wording, yet.

If we changed it to view, we could also also rename the concept to `range-reconstructible-view` or `view-reconstructible-from-range` (and `view-reconstructible-from-pair`) which sounds a little better than `range-reconstructible-range` 😆